### PR TITLE
Make K8s target provider apply deployment if container spec changes in Solution

### PR DESCRIPTION
**Pull request by LeiQL**
Fix issue -#240

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go [hunk 1]
```diff
@@ -630,15 +630,15 @@ func (i *K8sTargetProvider) GetValidationRule(ctx context.Context) model.Validat
 			RequiredMetadata:      []string{},
 			OptionalMetadata:      []string{},
 			ChangeDetectionProperties: []model.PropertyDesc{
-				{Name: model.ContainerImage, IgnoreCase: true, SkipIfMissing: false},
+				{Name: "container.*", IgnoreCase: true, SkipIfMissing: false},
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go:633]: Please consider to include a test for this change.

Comment (LeiQL) [reply api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go:633]: Manual tested. Do we need an integration for this minor change?

Comment (msftcoderdjw) [reply api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go:633]: Yes. My concern is someone change some logic in the future and you cannot know this feature is broken if we don't have any gates for this scenario.

Comment (Haishi2016) [reply api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go:633]: Agree with test cases.

Comment (LeiQL) [reply api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go:633]: Test added and passed.

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**